snmp/tests/test_utils.py

```diff
@@ -7,43 +7,49 @@
 from datadog_checks.snmp.pysnmp_types import MibViewController, ObjectIdentity, ObjectName, ObjectType, SnmpEngine
 
 
-@pytest.mark.unit
-def test_oid():
-    # type: () -> None
-    oid = OID((1, 3, 6, 1, 2, 1, 0))
-    assert oid.as_tuple() == (1, 3, 6, 1, 2, 1, 0)
-    assert repr(oid) == "OID('1.3.6.1.2.1.0')"
-    assert str(oid) == '1.3.6.1.2.1.0'
-
-
 @pytest.mark.unit
 @pytest.mark.parametrize(
-    'value, expected_tuple',
+    'value',
     [
-        pytest.param((1, 3, 6, 1, 2, 1, 0), (1, 3, 6, 1, 2, 1, 0), id='tuple'),
-        pytest.param('1.3.6.1.2.1.0', (1, 3, 6, 1, 2, 1, 0), id='string'),
-        pytest.param((1, '3', 6, '1', 2, 1, '0'), (1, 3, 6, 1, 2, 1, 0), id='mixed-tuple'),
-        pytest.param('.1.3.6.1.2.1.0', (1, 3, 6, 1, 2, 1, 0), id='string-with-leading-dot'),
-        pytest.param(ObjectName((1, 3, 6, 1, 2, 1, 0)), (1, 3, 6, 1, 2, 1, 0), id='object-name-tuple'),
-        pytest.param(ObjectName('1.3.6.1.2.1.0'), (1, 3, 6, 1, 2, 1, 0), id='object-name-string'),
+        pytest.param((1, 3, 6, 1, 2, 1, 1, 2), id='int-tuple'),
+        pytest.param(('1', '3', '6', '1', '2', '1', '1', '2'), id='string-tuple',),
+        pytest.param((1, '3', 6, '1', 2, 1, '1', 2), id='mixed-tuple'),
+        pytest.param('1.3.6.1.2.1.1.2', id='string'),
+        pytest.param('.1.3.6.1.2.1.1.2', id='string-leading-dot'),
+        pytest.param(ObjectName((1, 3, 6, 1, 2, 1, 1, 2)), id='object-name-tuple'),
+        pytest.param(ObjectName('1.3.6.1.2.1.1.2'), id='object-name-string'),
+        pytest.param(ObjectIdentity((1, 3, 6, 1, 2, 1, 1, 2)), id='object-identity-tuple'),
+        pytest.param(ObjectIdentity('1.3.6.1.2.1.1.2'), id='object-identity-string'),
+        pytest.param(ObjectType(ObjectIdentity((1, 3, 6, 1, 2, 1, 1, 2))), id='object-type'),
         pytest.param(
-            lambda controller: ObjectIdentity((1, 3, 6, 1, 2, 1, 0)).resolveWithMib(controller),
-            (1, 3, 6, 1, 2, 1, 0),
+            lambda controller: ObjectIdentity('SNMPv2-MIB', 'sysObjectID').resolveWithMib(controller),
             id='resolved-object-identity',
         ),
         pytest.param(
-            lambda controller: ObjectType(ObjectIdentity((1, 3, 6, 1, 2, 1, 0))).resolveWithMib(controller),
-            (1, 3, 6, 1, 2, 1, 0),
+            lambda controller: ObjectType(ObjectIdentity('SNMPv2-MIB', 'sysObjectID')).resolveWithMib(controller),
             id='resolved-object-type',
         ),
     ],
 )
-def test_oid_parse_valid(value, expected_tuple):
-    # type: (Any, Tuple[int, ...]) -> None
-    mib_view_controller = MibViewController(SnmpEngine().getMibBuilder())
+def test_oid_from_resolvable_parts(value):
```

Comment:
This is a merge of `test_oid` and `test_oid_parse_valid`, which a few extra cases since we now also supported detecting the OID parts from unresolved `ObjectIdentity`/`ObjectType` instances.